alacritty/src/event.rs

```diff
@@ -431,10 +445,8 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
     #[inline]
     fn push_search(&mut self, c: char) {
         if let Some(regex) = self.search_state.regex.as_mut() {
-            // Prevent previous search selections from sticking around when not in vi mode.
-            if !self.terminal.mode().contains(TermMode::VI) {
-                self.terminal.selection = None;
-            }
+            // Clear selection so we do not obstruct any matches.
+            self.terminal.selection = None;
```

Comment:
We should clear it only for normal search, and keep it in Vi mode, so you can expand selection by searching, like you can in vim.

Reply:
Hmm, that's a good point and something I definitely want to be able to do. But the active match problematic is the same for vi mode, isn't it?

Might just better to never reset the selection then, so it behaves uniformly at least?

Reply:
Or to keep Vi cursor over active match while typing?

Reply:
What would that achieve?

Reply:
Nothing.

Anyway, I'm not sure if we want to have it for normal mode, for Vi mode I think it's not really a problem since your selection updates while you're typing, so it's more clear what is active, etc, and it seems to match what Vi does, since it keeps selection.